tests/PHPStan/Type/Php/RangeFunctionReturnTypeExtensionTest.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace PHPStan\Type\Php;
+
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Type\ArrayType;
+use PHPStan\Type\FloatType;
+use PHPStan\Type\IntegerType;
+use PHPStan\Type\StringType;
+use PHPUnit\Framework\MockObject\MockObject;
+
+class RangeFunctionReturnTypeExtensionTest extends \PHPStan\Testing\TestCase
```

Comment:
Hi, this isn't at all how dynamic return type extensions are tested. Please add new dataProvider to `NodeScopeResolverTest::testFileAsserts` and test it there. THank you.